site/learn/expert-advice/authentication/webauthn-explained.md

```diff
@@ -0,0 +1,478 @@
+
+In order to use our store, people need to log in to perform tasks like checking their order status, adding things to their cart, and placing orders. Logging in will tie the user's identity to other data like their mailing address and credit card. If you want to let folks log in securely, what are your options?
+
+### Passwords
+
+Passwords are a solid choice, but can be forgotten, stolen or re-used. In fact, over 11B username/password pairs are publicly available on the site HaveIBeenPwned. User's aren't really fans of remembering passwords either, and insecurely re-use them across different systems.
+
+This means that user accounts on your system don't simply depend on your security posture, but on the posture of all the other sites that the user has registered for with the same password.
+
+Doh!
+
+### Password Managers
+
+Well, what about password managers?
+
+The user experience can be tough when dealing with password managers. If you're using a password manager that integrates into the browser, any non-standard login form UX can be difficult for it to handle.
+
+If a password manager is centralized, it is now a high-value target. If it is not, then users can't login to accounts from more than one device.
+
+### MFA
+
+If you add an additional factor of authentication (i.e. text the user a code, send them an email with a code, or require them to enter some code based on some algorithm using something like Google Authenticator), you can increase security. These additional factors of the authentication.
+
+They add additional assurance about who the user is during an authentication process. However, the user experience is tough. It can be a hassle to force someone to go to a different device to get a code. Depending on the method, it can also be insecure. For example, SMS codes can be phished.
+
+### Single Sign-on
+
+Single sign on is another option, which doesn't have these same security risks, or rather, it centralizes them. Which makes some sense; I pay more attention to my Google account than to the many random accounts I've created on other sites.
+
+However, single sign-on has a business risk. You are now depending on a third party to allow your users to access your website.
+
+## WebAuthn
+
+An alternative is to use WebAuthn.
+
+With WebAuthn, the server only stores public keys. Public keys are designed to be public, and so aren't much of a target for any attacker. The private keys, which are the other half of the credential, are stored in a distributed manner across authenticators held by each user. 
+
+WebAuthn is built into all browsers and is phishing-resistant because it's tied to a specific piece of hardware or software. The user not only has to have access to that hardware or software, but also can't share that access with anyone else. WebAuthn is only available on HTTPS sites, and has protection against domain name mismatches. These all combine to offer a more secure experience than MFA or a password manager. 
```

Comment:
I think the TLS requirement and domain checking contributes more to the phishing resistance than being tied to hardware/software. The hardware attachment means that someone needs to be in possession of the device in order to authenticate, but my understanding is that phishing typically involves a user on their own device.

The domain name (Relying Party ID) is used to scope the credentials and means that, for example, your credential for google.com can't be used on goagle.com (or some other website with a similar name).